node/src/rpc_effectful/rpc_effectful_effects.rs

```diff
@@ -804,6 +810,16 @@ fn compute_node_status<S: Service>(store: &mut Store<S>) -> RpcNodeStatus {
             transaction_candidates: state.transaction_pool.candidates.transactions_count(),
         },
         current_block_production_attempt,
+        resources_status: RpcNodeStatusResources {
+            p2p_malloc_size: {
+                let mut set = BTreeSet::new();
+                let fun = move |ptr: *const c_void| !set.insert(ptr.addr());
```

Comment:
@vlad9486 the compiler is complaining because of this line, error is:

```
error[E0658]: use of unstable library feature 'strict_provenance'
   --> node/src/rpc_effectful/rpc_effectful_effects.rs:816:[69](https://github.com/openmina/openmina/actions/runs/12887689269/job/35930919449?pr=1002#step:6:70)
    |
816 |                 let fun = move |ptr: *const c_void| !set.insert(ptr.addr());
    |                                                                     ^^^^
    |
    = note: see issue #95228 <https://github.com/rust-lang/rust/issues/95228> for more information
 ```
 
 But I see it has been stabilized a few months ago, maybe a more recent release of the compiler already includes that feature?

Reply:
@vlad9486 https://www.infoworld.com/article/3732079/rust-1-84-introduces-strict-provenance-apis.html

1.84 includes it, so we just need to upgrade

Reply:
Making that change in another PR, you can then rebase this PR on top of `develop` after I have merged it

Reply:
#1030 merged, please rebase